### ISSUE TITLE
Two columns layout using flex breaks on page wrap

### ISSUE THREAD

**Issue by Fuzy** [diegomura/react-pdf#430]
**OS:** Ubuntu 18.04.1 LTS

**React-pdf version:**  1.1.0

**Description:** I use rows with two colums. Page wrap is active. If a row is split among two pages,  layout of columns is broken. One of pages has correct width, other is wrong. If I define width by percentage it works correctly.

**How to replicate issue including code snippet (if applies):**

```
import React from 'react';
import ReactPDF, {
  Document,
  Page,
  View,
  Text,
  Font,
  StyleSheet,
} from '../../dist/react-pdf.es.js';

const styles = StyleSheet.create({
  page: {
    padding: 10,
  },
  text: {
    margin: 10,
    fontFamily: 'Oswald',
    textAlign: 'justify',
  },
  h2: {
    display: 'block',
    fontSize: 30,
    marginTop: 1,
    marginBottom: 1,
    marginLeft: 0,
    marginRight: 0,
    padding: 10,
  },
  row: {
    flex: 1,
    flexDirection: 'row',
    flexGrow: 1,
  },
  left: {
    // width: '33%',//<- working alternative
    flexGrow: 0,
    flexShrink: 1,
    flexBasis: 200,
  },

  right: {
    padding: 5,
    // width: '66%', //<- working alternative
    flexShrink: 1,
    flexGrow: 2,
  },
});

Font.register(
  'https://fonts.gstatic.com/s/oswald/v13/Y_TKV6o8WovbUd3m_X9aAA.ttf',
  { family: 'Oswald' },
);

const doc = (
  <Document>
    <Page style={styles.page} wrap size="A4">
      <View style={[styles.row, { height: 700 }]}>
        <View debug style={styles.left}>
          <Text style={styles.h2}>Headline1</Text>
        </View>
        <View debug style={styles.right}>
          <Text>
            ...
          </Text>
        </View>
      </View>
      <View style={styles.row}>
        <View debug style={styles.left}>
          <Text style={styles.h2}>Headline3</Text>
        </View>
        <View debug style={styles.right}>
          <Text>
            Lorem Ipsum is simply dummy text of the printing and typesetting
            industry. Lorem Ipsum has been the industry's standard dummy text
            ever since the 1500s, when an unknown printer took a galley of type
            and scrambled it to make a type specimen book. It has survived not
            only five centuries, but also the leap into electronic typesetting,
            remaining essentially unchanged. It was popularised in the 1960s
            with the release of Letraset sheets containing Lorem Ipsum passages,
            and more recently with desktop publishing software like Aldus
            PageMaker including versions of Lorem Ipsum.
          </Text>
        </View>
      </View>
    </Page>
  </Document>
);
```

ReactPDF.render(doc, `${__dirname}/two-columns-page-wrap.pdf`);

[two-columns-page-wrap.pdf](https://github.com/diegomura/react-pdf/files/2702347/two-columns-page-wrap.pdf)

https://react-pdf.org/repl?code=3187b0760ce02e004d09e01b02984605e18194a4e5600b6592803a6002764043299002806f00a06180076a0736402e1859b369c00988809660bbf008c001800d2b18017c95b3a003ca3f414202db50a5d26cc5cad80337050018b57de313c7e01c803c8400eed5108d7ea425a50008288e25c606e005600aed0e296f001ca6aca04004cba16301210ec88d42e30ae00468820c000d6294230d6605058e2005e7c3000cce67586c692002a20ecb2816cbd2660004220505020fa2339e3920032c8963a30dd064613004a11049bdbc2d46292d230f281696c14205ed97596289a8bcfaf0022e254c050e2e0373dcbcb5210bd909a0038b03deaa408a0364f21001e991302f3884450021b83a1d002900551001e002d3a240142a85c60be3a050c0b471000dd903970542615b51bd55e840a248aab0ab2bd26d40838820fc0c9c9b6696500ab8472469dce527e0015879a8f4663b16e001b01b090a18093c95e4a752a4b4c43d319ff567b2f9040158085571e473a10f694dc942a0025001b998cc5b0d948541334190147a3295c47283b0a5a88694020a42e34099c0723cd911064481bcbe1132399320eb22009a007d7e801a4006a06900003800ea206669400aa220ebe9eb000d00273504221522cd2ca0c63d41c4e62878cb7e571c39821b0e8120b04fa5430307a20630003e13f298987e02c5f4c846b9e72c48002b7190705c0a1d08c040a0b34e078151d10a1a8760e056990740002210800161839f3a8d86245b711902f0bf3c17f001b5ff540a30794d45c8843936001d965550005d1519094350f4330dc990529622e1b09fcff6fd08844a07a272463897e82158008dfc08acd327a200091a044708c06406462591513b406384e4598af0b4942d08c2b0910d88e2b8e8278bc0b34548e41318ba844b13f4fb260520dca120cf529c8f3509d28ce72603537480b0c96224cb200a22bc3b3ecd0b8cd3338f0aa4d21f898a5cc73b473324de3a48c8e485294e403a35234a8002df3829f302dd358f6312de3b2ab2a3722d2d8b4af2a5095929641f4180004954def18125483f4029e05c9ef7d026e08601012c181b14fdd8375fe1b5a83004445be076150120d6ae0aaba924111e22802878148180baaa17a81a2021a08714605298830116a2186cdace8bb5c341734da8c2db4efd1a6c5ac4a3a84641590a120b01804fc96ab83559420534bc221de8da60589dd3001e77a56c91e945a40100aa5a4602e17c14026f9bb6ddb21b6036ada204a01c729902dbc4580e61810c2a93f6a019cfdf2641807101f77b4a326aa2bafad809effb620a199164b9981f1a8099b9ac0671ea756600471a15630b465ed89605f0201003ecfc50702bebe790140fe7b8c071180116207da2e1e45cdbba8490695407dc69c41a626dc78027aa42e7e5d807c341d821962428051f7b9f7a91991c7034e4080758c5b13b6602a01d9f6e685ad673bc5120e022048340f7280838f66d1ba7afeb06deb38080200fcd19d659fe7bab2fc5c7ca07d78b82158880aa39820f616272925020691b6361f0a8181c2416603094ec2e5cba9df1e000596a105d872460110588241b461895c0341e9ceeee9ef484864aef3d2bf3308850017a45f32233ec80189a95bcf7927b3e1dccc0f60d03f8af93e2d828c8f84cbc61bc4799139e7816180858620000

**Comment by diegomura** [comment]
Sorry for the delay. I was out for some weeks 😄 

I was testing this scenario and it seems that it's caused by the `flexGrow: 0` on the `left` column. It's unclear yet why this only happens when wrapping 🤔 
Also the margin on the `h2` is causing the right column to think it has a bit more of space to render the text. I would always recommend to add paddings to the text's container and not to the text itself.

I will try to debug a bit more and see if I can find a fix for this

**Comment by diegomura** [comment]
Fixed in 2.0. Check [here](https://react-pdf.org/repl?code=3187b0760ce02e004d09e01b02984605e18194a4e5600b6592803a6002764043299002806f00a06180076a0736402e1859b369c00988809660bbf008c001800d2b18017c95b3a003ca3f414202db50a5d26cc5cad80337050018b57de313c7e01c803c8400eed5108d7ea425a50008288e25c606e005600aed0e296f001ca6aca04004cba16301210ec88d42e30ae00468820c000d6294230d6605058e2005e7c3000cce67586c692002a20ecb2816cbd2660004220505020fa2339e3920032c8963a30dd064613004a11049bdbc2d46292d230f281696c14205ed97596289a8bcfaf0022e254c050e2e0373dcbcb5210bd909a0038b03deaa408a0364f21001e991302f3884450021b83a1d002900551001e002d3a240142a85c60be3a050c0b471000dd903970542615b51bd55e840a248aab0ab2bd26d40838820fc0c9c9b6696500ab8472469dce527e0015879a8f4663b16e001b01b090a18093c95e4a752a4b4c43d319ff567b2f9040158085571e473a10f694dc942a0025001b998cc5b0d948541334190147a3295c47283b0a5a88694020a42e34099c0723cd911064481bcbe1132399320eb22009a007d7e801a4006a06900003800ea206669400aa220ebe9eb000d00273504221522cd2ca0c63d41c4e62878cb7e571c39821b0e8120b04fa5430307a20630003e13f298987e02c5f4c846b9e72c48002b7190705c0a1d08c040a0b34e078151d10a1a8760e056990740002210800161839f3a8d86245b711902f0bf3c17f001b5ff540a30794d45c8843936001d965550005d1519094350f4330dc990529622e1b09fcff6fd08844a07a272463897e82158008dfc08acd327a200091a044708c06406462591513b406384e4598af0b4942d08c2b0910d88e2b8e8278bc0b34548e41318ba844b13f4fb260520dca120cf529c8f3509d28ce72603537480b0c96224cb200a22bc3b3ecd0b8cd3338f0aa4d21f898a5cc73b473324de3a48c8e485294e403a35234a8002df3829f302dd358f6312de3b2ab2a3722d2d8b4af2a5095929641f4180004954def18125483f4029e05c9ef7d026e08601012c181b14fdd8375fe1b5a83004445be076150120d6ae0aaba924111e22802878148180baaa17a81a2021a08714605298830116a2186cdace8bb5c341734da8c2db4efd1a6c5ac4a3a84641590a120b01804fc96ab83559420534bc221de8da60589dd3001e77a56c91e945a40100aa5a4602e17c14026f9bb6ddb21b6036ada204a01c729902dbc4580e61810c2a93f6a019cfdf2641807101f77b4a326aa2bafad809effb620a199164b9981f1a8099b9ac0671ea756600471a15630b465ed89605f0201003ecfc50702bebe790140fe7b8c071180116207da2e1e45cdbba8490695407dc69c41a626dc78027aa42e7e5d807c341d821962428051f7b9f7a91991c7034e4080758c5b13b6602a01d9f6e685ad673bc5120e022048340f7280838f66d1ba7afeb06deb38080200fcd19d659fe7bab2fc5c7ca07d78b82158880aa39820f616272925020691b6361f0a8181c2416603094ec2e5cba9df1e000596a105d872460110588241b461895c0341e9ceeee9ef484864aef3d2bf3308850017a45f32233ec80189a95bcf7927b3e1dccc0f60d03f8af93e2d828c8f84cbc61bc4799139e7816180858620000). Had to add a small change though in the `row` style. You cannot set both `flex: 1` and setup a fixed height.
